atacworks/dl4atac/utils.py

```diff
@@ -525,6 +554,41 @@ def get_intervals(sizesfile, intervalsize, out_dir, val=None,
         df_to_bed(holdout, holdout_file_path)
         return train_file_path, val_file_path, holdout_file_path
 
+    elif regions is not None:
+        # If given regions is a file, then just return the file path
+        if regions.endswith(".bed"):
+            return regions
+        else:
+            final_intervals = pd.DataFrame()
+            regions = regions.strip("[]").split(",")
+            for region in regions:
+                # If regions are specified with intervals like chr1:0-50
+                # Then split the region into chrom and it's range.
+                if region.find(":") != -1:
+                    chrom, chrom_range = region.split(":")
+                    chrom_range = chrom_range.split("-")
+                    chrom_range = [int(value) for value in chrom_range]
+                    chrom_sizes = sizes[sizes['chrom'] == chrom]
+                    intervals = _tile_region_intervals(
+                        chrom_sizes.iloc[0,0],
```

Comment:
Why do we need to use the sizes file here? The name of the chromosome is already supplied by the user as part of `--regions` and is stored in the `chrom` variable.

Reply:
Good catch, left over from prev changes. Removed it.